### PR TITLE
Add solution

### PR DESCRIPTION
**Pull request by PavloSich**
[DEMO LINK](https://kapidastra.github.io/react_list-of-goods-js/)

### CODE REVIEW

**Review by volodymyr-soltys97 — CHANGES_REQUESTED** [review]
Good job 👍 
Let's improve your code
Reverse does not work correctly
image
image
image
image

Also you need to fix the failed test, if you need a help feel free in the fe_chat

File: src/App.jsx [hunk 1]
```diff
@@ -14,33 +16,115 @@ export const goodsFromServer = [
...
+  const [buttonClassSortAlpha, setButtonClassSortAlpha] =
+    useState('is-info is-light');
+  const [buttonClassSortByLength, setButtonClassSortByLength] = useState(
+    'is-success is-light',
+  );
+  const [buttonClassReverse, setButtonClassReverse] = useState(
+    'is-warning is-light',
+  );
+  const [buttonClassReset, setButtonClassReset] =
+    useState('is-danger is-light');
```
Comment (vadiimvooo) [inline src/App.jsx:52]: Looks very strange, just create state for sortField and store sort value here. Later you can reuse it with classnames library to change color of button.

File: src/App.jsx [hunk 2]
```diff
@@ -14,33 +16,115 @@ export const goodsFromServer = [
...
+      return [...goods].sort((good1, good2) => good1.localeCompare(good2));
+    case SORT_BY_LENGTH:
+      return [...goods].sort((good1, good2) => good1.length - good2.length);
+    case REVERSE_LIST:
+      return [...goodsFromServer].reverse();
+    case RESET:
+      return [...goodsFromServer];
+    default:
+      return goods;
+  }
```
Comment (vadiimvooo) [inline src/App.jsx:36]: The function should be rewrited. Please sort it and after sort make reverse of array in case reverse state is true.

**Review by vadiimvooo — CHANGES_REQUESTED** [review]
Good job, but you have to make changes to your code to pass tests. Please feel free to ask for help in fe_chat.

File: src/App.jsx [hunk 3]
```diff
@@ -14,33 +16,135 @@ export const goodsFromServer = [
...
+
+export const App = () => {
+  const [sortingType, setSortingType] = useState(null);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const buttonClick = buttonId => {
+    switch (buttonId) {
+      case SORT_ALPHABETICALLY:
+        setSortingType(SORT_ALPHABETICALLY);
+        setIsReversed(false);
```
Comment (anastasiiavorobiova) [inline src/App.jsx:57]:
```suggestion

```

File: src/App.jsx [hunk 4]
```diff
@@ -14,33 +16,135 @@ export const goodsFromServer = [
...
+        break;
+      case SORT_BY_LENGTH:
+        setSortingType(SORT_BY_LENGTH);
+        setIsReversed(false);
```
Comment (anastasiiavorobiova) [inline src/App.jsx:61]:
```suggestion

```

File: src/App.jsx [hunk 5]
```diff
@@ -14,33 +16,135 @@ export const goodsFromServer = [
...
+  const resetGoods = () => {
+    setSortingType(null);
+    setIsReversed(false);
+  };
+
+  const isDifferent = () => {
+    return !visibleVegetables.every(
+      (item, index) => item === goodsFromServer[index],
+    );
+  };
```
Comment (anastasiiavorobiova) [inline src/App.jsx:84]:
```suggestion

```

File: src/App.jsx [hunk 6]
```diff
@@ -14,33 +16,135 @@ export const goodsFromServer = [
...
+
+  const visibleVegetables = isReversed
+    ? [...goodsFromServer].reverse()
+    : [...goodsFromServer];
```
Comment (anastasiiavorobiova) [inline src/App.jsx:88]:
```suggestion

```

File: src/App.jsx [hunk 7]
```diff
@@ -14,33 +16,135 @@ export const goodsFromServer = [
...
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const preparedGoods = sortingType
```
Comment (anastasiiavorobiova) [inline src/App.jsx:90]:
```suggestion
  const preparedGoods = (sortingType || isReversed)
```

File: src/App.jsx [hunk 8]
```diff
@@ -14,33 +16,135 @@ export const goodsFromServer = [
...
+    ? getPreparedGoodsFromServer(visibleVegetables, sortingType, isReversed)
```
Comment (anastasiiavorobiova) [inline src/App.jsx:91]:
```suggestion
    ? getPreparedGoodsFromServer([...goodsFromServer], sortingType, isReversed)
```

File: src/App.jsx [hunk 9]
```diff
@@ -14,33 +16,135 @@ export const goodsFromServer = [
...
+    : visibleVegetables;
```
Comment (anastasiiavorobiova) [inline src/App.jsx:92]:
```suggestion
    : [...goodsFromServer];
```

File: src/App.jsx [hunk 10]
```diff
@@ -14,33 +16,135 @@ export const goodsFromServer = [
...
+          type="button"
+          className={cn('button', {
+            'is-warning': isReversed,
+            'is-warning is-light': !isReversed,
+          })}
+        >
+          Reverse
+        </button>
+
+        {(isDifferent() || isReversed || sortingType) && (
```
Comment (anastasiiavorobiova) [inline src/App.jsx:130]:
```suggestion
        {(isReversed || sortingType) && (
```

**Review by anastasiiavorobiova — CHANGES_REQUESTED** [review]
Great work! Added a few comments to fix

**Review by anastasiiavorobiova — APPROVED** [review]
Great work!

**Pull request closed without merging**